test/Mono.Linker.Tests.Cases/DataFlow/MethodParametersDataFlow.cs

```diff
@@ -0,0 +1,197 @@
+﻿using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	[KeptMember(".ctor()")]
+	public class MethodParametersDataFlow
+	{
+		public static void Main ()
+		{
+			var instance = new MethodParametersDataFlow ();
+
+			DefaultConstructorParameter (typeof (TestType));
```

Comment:
I'm not a big fan of loading up a single test case with multiple usage scenarios.  It's easy to get yourself into a situation where your test isn't providing the coverage that you think it is because multiple things lead to the same thing being marked and if one of those mechanisms broke, you'd never know.

After the first `[DynamicallyAccessedMembers(DynamicallyAccessedMemberKinds.DefaultConstructor)]` is processed any additional uses of that attribute with the same `DynamicallyAccessedMemberKinds` value and the same Type are not providing much in the way of test coverage.

I put a break point in `EnqueueMethod` for the `TestType.ctor()` and `RequireDynamicallyAccessedMembers` is marking it >3 times.  If every usage after the first broke you'd never know.

Reply:
Or maybe I'm misunderstanding and the primary goal of these tests isn't whether or not `TestType` ctors are marked?  But rather that the `UnrecognizedReflectionAccessPattern` are reported correctly?

Reply:
The purpose of the test is to validate the `UnrecognizedReflectionAccessPattern` behavior. These will eventually turn into warnings, so maybe we should write the tests later on validating the warnings instead, but I prefer specific approach like this better myself.
The marking of the `TestType.ctor` is a "side-effect" of these tests. In fact I considered if I should add a way for the test to opt-out of the `Kept` validation as nothing in this test actually needs that.

Reply:
I added comment to the test.

What do you think about introducing a way to disable the `Kept` validation for specific tests?

Reply:
👍